src/engine/plugins/tile.cpp

```diff
@@ -380,226 +387,218 @@ Status TilePlugin::HandleRequest(const std::shared_ptr<datafacade::BaseDataFacad
                 }
                 else
                 {
-                    incoming_edges[edge.v] = {edge.forward_segment_id.id};
+                    directed_graph[edge.u].push_back({edge.v, edge.forward_segment_id.id});
                 }
-
-                edge_based_node_info[edge.forward_segment_id.id] = {
-                    edge.u, edge.v, edge.packed_geometry_id};
+                edge_based_node_info[edge.forward_segment_id.id] = {true, edge.packed_geometry_id};
             }
-            // Same as previous block, but everything flipped
-            if (edge.reverse_segment_id.enabled &&
-                edge_based_node_info.count(edge.reverse_segment_id.id) == 0)
+            if (edge.reverse_segment_id.enabled)
             {
-                auto f = outgoing_edges.find(edge.v);
-                if (f != outgoing_edges.end())
-                {
-                    f->second.push_back(edge.reverse_segment_id.id);
-                }
-                else
+                if (directed_graph.count(edge.v) == 0)
                 {
-                    outgoing_edges[edge.v] = {edge.reverse_segment_id.id};
-                }
-
-                f = incoming_edges.find(edge.u);
-                if (f != incoming_edges.end())
-                {
-                    f->second.push_back(edge.reverse_segment_id.id);
+                    directed_graph[edge.v] = {{edge.u, edge.reverse_segment_id.id}};
                 }
                 else
                 {
-                    incoming_edges[edge.u] = {edge.reverse_segment_id.id};
+                    directed_graph[edge.v].push_back({edge.u, edge.reverse_segment_id.id});
                 }
-
-                // Save info about this edge-based-node, note reversal from forward
-                // block above.
-                edge_based_node_info[edge.reverse_segment_id.id] = {
-                    edge.v, edge.u, edge.packed_geometry_id};
+                edge_based_node_info[edge.reverse_segment_id.id] = {false, edge.packed_geometry_id};
             }
         }
 
-        // Now, for every edge-based-node that we discovered (edge-based-nodes are sources
-        // and targets of turns).  EBN is short for edge-based-node
-        std::vector<NodeID> first_geometry, second_geometry;
+        // Now, scan over our adjacency list
+        // For every edge A:
+        //   Look at the outgoing edges from A
+        //   If the outgoing edge has a different edge_based_node_id
+        //     This is a turn.  Find it in the CH
+        //     Get the edge data
+        //     Subtract the road weights from the edge weight
+        //     Calculate angles, add to tile.
         std::vector<contractor::QueryEdge::EdgeData> unpacked_shortcut;
-        std::vector<EdgeWeight> forward_weight_vector;
-        for (const auto &source_ebn : edge_based_node_info)
+        std::vector<EdgeWeight> first_weight_vector;
```

Comment:
is there a way we can choose more meaningful names here? firstnode/firstedge seem a bit strange for me, if we actually loop over many of them.